File: Language/C#/Book/LINQ实战/第06章 LINQ to SQL入门/Chapter06to08.VB/Common/SampleClasses/Ch7/Author.vb
Imports System.Data.Linq.Mapping
Imports System.Data.Linq

Namespace SampleClasses.Ch7

    <Table(Name:="dbo.Author")> _
    Public Class Author

        Private _ID As System.Guid
        <Column(Storage:="_ID", Name:="ID", DbType:="UniqueIdentifier NOT NULL", IsPrimaryKey:=True, CanBeNull:=False)> _
        Public Property ID() As System.Guid
            Get
                Return _ID
            End Get
            Set(ByVal value As System.Guid)
                _ID = value
            End Set
        End Property

        Private _LastName As String
        <Column(Name:="LastName", DbType:="VarChar(50) NOT NULL", CanBeNull:=False, UpdateCheck:=UpdateCheck.Never)> _
        Public Property LastName() As String
            Get
                Return _LastName
            End Get
            Set(ByVal value As String)
                _LastName = value
            End Set
        End Property

        Private _FirstName As String
        <Column(Name:="FirstName", DbType:="VarChar(30) NOT NULL", CanBeNull:=False, UpdateCheck:=UpdateCheck.Never)> _
        Public Property FirstName() As String
            Get
                Return _FirstName
            End Get
            Set(ByVal value As String)
                _FirstName = value
            End Set
        End Property

        Private _WebSite As String
        <Column(Name:="WebSite", DbType:="VarChar(200)", UpdateCheck:=UpdateCheck.Never)> _
        Public Property WebSite() As String
            Get
                Return _WebSite
            End Get
            Set(ByVal value As String)
                _WebSite = value
            End Set
        End Property

        Private _TimeStamp As Byte()
        <Column(Name:="TimeStamp", DbType:="rowversion NOT NULL", IsDbGenerated:=True, IsVersion:=True, CanBeNull:=False, UpdateCheck:=UpdateCheck.Always)> _
        Public Property TimeStamp() As Byte()
            Get
                Return _TimeStamp
            End Get
            Set(ByVal value As Byte())
                _TimeStamp = value
            End Set
        End Property

        Private _BookAuthors As EntitySet(Of BookAuthor)
        <Association(Name:="FK_BookAuthor_Author", Storage:="_BookAuthors", OtherKey:="Author", ThisKey:="ID")> _
        Public Property BookAuthors() As EntitySet(Of BookAuthor)
            Get
                Return _BookAuthors
            End Get
            Set(ByVal value As EntitySet(Of BookAuthor))
                _BookAuthors.Assign(value)
            End Set
        End Property

    End Class
End Namespace

File: Language/C#/Book/LINQ实战/第06章 LINQ to SQL入门/Chapter06to08.CSharp/Common/SampleClasses/Ch7/Author.cs
using System;
using System.Collections.Generic;
using System.Data.Linq;
using System.Data.Linq.Mapping;
using System.Linq;
using System.Text;

namespace LinqInAction.Chapter06to08.Common.SampleClasses.Ch7
{
    [Table(Name = "dbo.Author")]
    public class Author
    {
        private System.Guid _ID;
        [Column(Storage = "_ID", Name = "ID", DbType = "UniqueIdentifier NOT NULL", IsPrimaryKey = true, CanBeNull = false)]
        public System.Guid ID { get { return _ID; } set { _ID = value; } }
        [Column(Name = "LastName", DbType = "VarChar(50) NOT NULL", CanBeNull = false, UpdateCheck = UpdateCheck.Never)]
        public string LastName { get; set; }
        [Column(Name = "FirstName", DbType = "VarChar(30) NOT NULL", CanBeNull = false, UpdateCheck = UpdateCheck.Never)]
        public string FirstName { get; set; }
        [Column(Name = "WebSite", DbType = "VarChar(200)", UpdateCheck = UpdateCheck.Never)]
        public string WebSite { get; set; }
        [Column(Name = "TimeStamp", DbType = "rowversion NOT NULL", IsDbGenerated = true, IsVersion = true, CanBeNull = false, UpdateCheck = UpdateCheck.Always)]
        public byte[] TimeStamp { get; set; }

        private EntitySet<BookAuthor> _BookAuthors = null;

        [Association(Name = "FK_BookAuthor_Author", Storage = "_BookAuthors", OtherKey = "Author", ThisKey = "ID")]
        public EntitySet<BookAuthor> BookAuthors
        {
            get
            {
                return this._BookAuthors;
            }
            set
            {
                this._BookAuthors.Assign(value);
            }
        }
    }
}